File: Backend/src/main/java/com/bookstore/entity/Order.java
package com.bookstore.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("orders")
public class Order {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private Long distributorId; // Distributor ID (from passcode)

    private Long sourcePasscodeId; // Source Passcode ID

    private String orderNo;

    private BigDecimal amount;

    private String status; // Pending, Paid, Refunded

    private String platform; // AppStore, GooglePay

    private String productId;

    // Subscription fields
    private String orderType; // onetime, subscription

    private String subscriptionPeriod; // monthly, quarterly, yearly

    private LocalDateTime subscriptionStartDate;

    private LocalDateTime subscriptionEndDate;

    private Boolean isAutoRenew;

    private LocalDateTime cancelDate;

    private String cancelReason;

    // Platform transaction fields
    private String originalTransactionId; // Apple/Google original transaction ID

    private String platformTransactionId;

    private String purchaseToken; // Google purchase token

    private String receiptData; // Apple receipt data

    private LocalDateTime verifiedAt; // When receipt was verified

    // Source tracking fields
    private Long sourceBookId; // From which book

    private String sourceEntry; // profile, reader

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}

File: App/lib/src/features/transaction_record/data/models/order.dart
import 'package:json_annotation/json_annotation.dart';

part 'order.g.dart';

@JsonSerializable()
class Order {
  final int id;
  final int userId;
  final int? distributorId;
  final int? sourcePasscodeId;
  final String orderNo;
  final double amount;
  final String status; // Pending, Paid, Refunded
  final String platform; // AppStore, GooglePay
  final String productId;
  final String? orderType; // onetime, subscription
  final String? subscriptionPeriod; // monthly, quarterly, yearly
  final DateTime? subscriptionStartDate;
  final DateTime? subscriptionEndDate;
  final bool? isAutoRenew;
  final DateTime? cancelDate;
  final String? cancelReason;
  final String? originalTransactionId;
  final String? platformTransactionId;
  final String? purchaseToken;
  final String? receiptData;
  final int? sourceBookId;
  final String? sourceEntry; // profile, reader
  final DateTime createTime;
  final DateTime? updateTime;

  Order({
    required this.id,
    required this.userId,
    this.distributorId,
    this.sourcePasscodeId,
    required this.orderNo,
    required this.amount,
    required this.status,
    required this.platform,
    required this.productId,
    this.orderType,
    this.subscriptionPeriod,
    this.subscriptionStartDate,
    this.subscriptionEndDate,
    this.isAutoRenew,
    this.cancelDate,
    this.cancelReason,
    this.originalTransactionId,
    this.platformTransactionId,
    this.purchaseToken,
    this.receiptData,
    this.sourceBookId,
    this.sourceEntry,
    required this.createTime,
    this.updateTime,
  });

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
  Map<String, dynamic> toJson() => _$OrderToJson(this);

  // Helper method to get status display text
  String get statusDisplay {
    switch (status) {
      case 'Pending':
        return 'Pending';
      case 'Paid':
        return 'Paid';
      case 'Refunded':
        return 'Refunded';
      default:
        return status;
    }
  }

  // Helper method to get subscription period display text
  String get subscriptionPeriodDisplay {
    switch (subscriptionPeriod) {
      case 'monthly':
        return 'Monthly';
      case 'quarterly':
        return 'Quarterly';
      case 'yearly':
        return 'Yearly';
      default:
        return subscriptionPeriod ?? 'One-time';
    }
  }
}
